src/test/java/org/candlepin/subscriptions/tally/TallySnapshotControllerIT.java

```diff
@@ -122,7 +122,8 @@ private void givenEventAtDay(int dayAt, double value) {
   }
 
   private void whenProduceHourlySnapshotsForOrg() {
-    controller.produceHourlySnapshotsForOrg(ORG_ID, new DateRange(start, end));
+    controller.produceHourlySnapshotsForOrg(
+        ORG_ID, new DateRange(clock.startOfCurrentHour(), clock.startOfCurrentHour().plusHours(1)));
```

Comment:
I'm trying to reason for the motivation of this change. 
Given today is 19 September 05:00
And the following events in database:
- event with timestamp at 18 September 13:00 (automatically record_date is set to today: 19 September 05:00)
- event with timestamp at 18 September 13:15 (automatically record_date is set to today: 19 September 05:00)

When we produce the hourly snapshots for org range between 18 September 13:00 <-> 14:00
Then no data is generated because the previous events record_date was set to a recent date (today) 19 September 05:00.

Therefore, we must produce the hourly snapshots for today's date range. 

I'm glad that this integration test spotted this scenario which I think is definitely a change of behavior that needs to be noted in the release notes (but I don't know if you have this kind of information when releasing) @kahowell @barnabycourt

Where I think the new behavior is correct (here we're interested in producing data from events that have been recently inserted in our database - aka the `record_date`-, instead of using the `timestamp` - which is when the event was made in a third-party system like prometheus).

My only concern is that taking into account events might take a while to be consumed from the kafka topic, users seeing an event in Prometheus at a concrete timestamp and might want to produce the tally report for this event using a date range within this concrete timestamp, will see that no data will be generated. However, I don't think this endpoint was ever meant to be used for this purpose, so it should be ok. 

FYI @himmAllRight @ntkathole